js/events.js

```diff
@@ -1,9 +1,25 @@
-import { createEventElement, createMultiDayEventElement, getFirstDayOfTheWeek, clearEvents, differenceBetweenTwoDatesInDays} from './utils.js';
+import { createEventElement, createMultiDayEventElement, getFirstDayOfTheWeek, clearEvents, differenceBetweenTwoDatesInDays, dateIsInRange} from './utils.js';
 import { getEventsFromLocalStorage } from './services.js';
 
 const HOUR_IN_MINUTES = 60;
+const doesEventOverlapWithOtherEvents = (eventCurrent) => {
+  const eventStartDateTime = new Date(eventCurrent.startDateTime);
+  const eventEndDateTime = new Date(eventCurrent.endDateTime);
+  const events = getEventsFromLocalStorage();
 
+  return events.filter((event) => {
+    const eventStart = new Date(event.startDateTime);
+    const eventEnd = new Date(event.endDateTime);
+    return (
+      (dateIsInRange(eventStart, eventEnd, eventStartDateTime) || 
+      dateIsInRange(eventStart, eventEnd, eventEndDateTime)) 
+   
+    );
+  });
+}
 const renderShortEvent = (event) => {
+  const overlappingEvents = doesEventOverlapWithOtherEvents(event);
```

Comment:
following up on what I said previous comment now:
- you get events from local storage to display them
- for each goes through events one by one
- for each event renderShortEvent function is called
- renderShortEvent calls doesEventOverlapWithOtherEvents
- doesEventOverlapWithOtherEvents again calls getEventsFromLocalStorage

so even though events do not really change, you repeat let's imagine network call that could be avoided.

it could be worthy to think either about getting events from local storage once before calling forEach (which you actually already do) and just pass it further on. or maybe to think about a bit different data structure, eg. once you get events you do calculations about how events overlaps and then just use it